pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -392,6 +431,45 @@ public static String normalize(String input, String form) {
     return StringUtils.split(input, delimiter);
   }
 
+  /**
+   * see String#split(String)
+   * @param input
+   * @param delimiter
+   * @return splits string on specified delimiter and returns an array.
+   */
+  @ScalarFunction
+  public static String split(String input, String delimiter, int index) {
```

Comment:
This is `splitPart` in presto (index starts with 1 though): https://prestodb.io/docs/current/functions/string.html
Consider making them consistent

Reply:
I have kept it consistent with existing pinot queries. In Presto, every string function operates on the expectation that String index starts with 1. 
Pinot's function on the other hand have String index 0.
If we want to change it, we should change all of them or none of them.

Also, changing them will break user queries who are already using existing functions.

Reply:
I see, let's keep the start index as 0.
I'd suggest adding it as `splitPart` so that it does not collide with `split` with limit. We can also add a `split` with 3 arguments which returns `StringUtils.split(input, delimiter, limit)`

Reply:
cool. renamed this function and added another one with limit.